src/types/timestamp.rs

```diff
@@ -1007,230 +973,212 @@ impl TimestampBuilder {
         }
         Ok(timestamp)
     }
+
+    /// Like [Self::build], but the fields provided for each time unit are understood
+    /// to be in UTC rather than in the local time of the specified offset (if there is one).
+    pub(crate) fn build_utc_fields_at_offset(
```

Comment:
🗺️  This seems like a specialized function for the binary reader, so I made it `pub(crate)` so that it's not in our public API. We can re-add it to the public API if there's a large outcry for it.

Reply:
It's true that it's only used by the binary reader within our crate. I do think it's generally useful; it's comparable to [this method](https://docs.rs/chrono/latest/chrono/struct.DateTime.html#method.from_utc) in the `chrono` crate, which lets you provide a set of UTC fields and an offset. That said, I'm ok with keeping it crate-visible for the time being.